meshapp/src/components/SignUp/StepOne.tsx

```diff
@@ -0,0 +1,374 @@
+            {/*Email*/}
+            <Grid item xs={12} sm={6}>
+              <EmailAddress />
+            </Grid>
+
+            {/*Password*/}
+            <Grid item xs={12} sm={6}>
+              <Password />
+            </Grid>
+
+            {/*Terms and Conditions/Email Opt-in*/}
+            <Grid
+              item
+              xs={12}
+              sm={6}
+              sx={{ color: "text.primary" }}
+              order={{ xs: 3, sm: 2 }}
+            >
+              <FormGroup>
+                <TermsAndConditions />
+
+                <EmailUpdates />
+              </FormGroup>
+            </Grid>
+
+            {/*Confirm Password*/}
+            <Grid item xs={12} sm={6} order={{ xs: 2, sm: 3 }}>
+              <ConfirmPassword />
+            </Grid>
+          </Grid>
+        </Grid>
+      </Grid>
+      <Divider sx={{ borderColor: "signUpDivider.main", opacity: 0.5 }} />
+    </Grid>
+  );
+}
+//---------------------------------------INPUTS-----------------------------------------------------------
+
+//first name text field
+function FirstName() {
+  return (
+    <CustomStandardTextField
+      label="First Name *"
+      id="first-name"
+      fieldName="firstName"
+      required={true}
+    />
+  );
+}
+
+//last name text field
+
+function LastName() {
+  return (
+    <CustomStandardTextField
+      label="Last Name *"
+      id="last-name"
+      required={true}
+      fieldName="lastName"
+    />
+  );
+}
+
+//nickname text field
+
+function Nickname() {
+  return (
+    <CustomStandardTextField
+      label="Nickname"
+      id="nickname"
+      fieldName="nickName"
+    />
+  );
+}
+
+//phone number text field
+
+function PhoneNumber() {
+  return (
+    <CustomStandardTextField
+      label="Phone Number"
+      id="phone-number"
+      fieldName="phoneNumber"
+      validators={{
+        pattern: (value) => {
+          let phoneNumRegex =
+            /^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]\d{3}[\s.-]\d{4}$/;
+          return phoneNumRegex.test(value) || "Invalid phone number format";
+        },
+      }}
+    />
+  );
+}
+
+//email address text field
+function EmailAddress() {
+  return (
+    <CustomStandardTextField
+      label="Email *"
+      id="email-address"
+      fieldName="email"
+      validators={{
+        pattern: (value) => {
+          let emailRegex =
+            /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+          return emailRegex.test(value) || "Invalid email address format";
+        },
+      }}
+    />
+  );
+}
+
+//password text field
+function Password() {
+  return (
+    <CustomPasswordTextField
+      fieldName="password"
+      id="password"
+      label="Password *"
+      validators={{
+        minLength: (value) => {
+          return value.length > 8 || "Password length needs to be at least 8.";
```

Comment:
Do you mean for this to be value.length >= 8? If someone enters a password of length 8, it still shows this error, so either the message needs to be changed to say it must be at least length 9, or the condition must be changed.